### PR TITLE
Initial cut for talosctl for FreeBSD (amd64/arm64)

### PR DESCRIPTION
**Pull request by lrosenman**
# Pull Request

## What? (description)
add FreeBSD support to talosctl 
## Why? (reasoning)
I want to be able to run talosctl from my base server for the bhyve VM's of Talos I have running
## Acceptance

Please use the following checklist:

- [ ] you linked an issue (if applicable)
- [ ] you included tests (if applicable)
- [ ] you ran conformance (`make conformance`)
- [ ] you formatted your code (`make fmt`)
- [ ] you linted your code (`make lint`)
- [ ] you generated documentation (`make docs`)
- [ ] you ran unit-tests (`make unit-tests`)

> See `make help` for a description of the available targets.

### CODE REVIEW

**Review by andrewrynhard — COMMENTED** [review]
`s/Freebsd/freebsd`

File: Dockerfile [hunk 1]
```diff
@@ -383,6 +383,19 @@ ARG GO_LDFLAGS
 ARG GOAMD64
 RUN --mount=type=cache,target=/.cache GOOS=windows GOARCH=amd64 GOAMD64=${GOAMD64} go build ${GO_BUILDFLAGS} -ldflags "${GO_LDFLAGS}" -o /talosctl-windows-amd64.exe
 
+FROM base AS talosctl-freebsd-amd64-build
+WORKDIR /src/cmd/talosctl
+ARG GO_BUILDFLAGS
+ARG GO_LDFLAGS
+ARG GOAMD64
+RUN --mount=type=cache,target=/.cache GOOS=Freebsd GOARCH=amd64 GOAMD64=${GOAMD64} go build ${GO_BUILDFLAGS} -ldflags "${GO_LDFLAGS}" -o /talosctl-freebsd-amd64.exe
```
Comment (andrewrynhard) [inline Dockerfile:391]: Drop `.exe`.

File: Dockerfile [hunk 2]
```diff
@@ -383,6 +383,19 @@ ARG GO_LDFLAGS
...
+
+FROM base AS talosctl-freebsd-arm64-build
+WORKDIR /src/cmd/talosctl
+ARG GO_BUILDFLAGS
+ARG GO_LDFLAGS
+RUN --mount=type=cache,target=/.cache GOOS=Freebsd GOARCH=arm64 go build ${GO_BUILDFLAGS} -ldflags "${GO_LDFLAGS}" -o /talosctl-freebsd-arm64.exe
```
Comment (andrewrynhard) [inline Dockerfile:397]: Drop `.exe`.

**Review by andrewrynhard — COMMENTED** [review]
you will need to update `.drone.jsonnet` as well to have the binary published. There are other `talosctl-*` entries you should be able to copy/paste.

**Comment by sauterp** [comment]
@lrosenman thanks for your contribution! If you'd like to make the CI-bot happy, you need to squash your commits into just one and change the commit message to the following format:

```
feat: build talosctl for FreeBSD

We add support for building talosctl for FreeBSD on
amd64 and arm64 architectures.
```

**Comment by lrosenman** [comment]
Can someone help me here?
The first commit was NOT signed, but the second one was.

❯ git rebase -i origin/master~2 master
error: gpg failed to sign the data
fatal: failed to write commit object
Could not apply 443620428... Address PR comments

talos on  HEAD (ab228e7) (REBASING 2/2) [+] is 📦 v0.0.1 via 🐹 v1.18.4 via  v18.6.0 on ☁️  (us-east-1) on ﴃ WhereTo - Prod took 20s
✖1 ❯

**Comment by rsmitty** [comment]
@lrosenman went ahead and squashed/amended your commit. one of us maintainers would have to sign it anyways, so 🤞 we're good to go now!

**Comment by rsmitty** [comment]
One more bit, I added [this](https://github.com/siderolabs/talos/pull/5906/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R402) step to actually copy out the built binaries. That should be everything I believe. Thx for working on this!

**Comment by lrosenman** [comment]
Thanks, @rsmitty.  I'll get the hang of this eventually.

**Comment by lrosenman** [comment]
I'm not sure what the build failure is.  If you need something from me, please ping.

**Comment by rsmitty** [comment]
will need https://github.com/siderolabs/go-blockdevice/pull/62 to be merged first.

**Pull request merged**